merlin/core/dispatch.py

```diff
@@ -715,3 +715,46 @@ def get_random_state():
     if cp:
         return cp.random.get_random_state()
     return np.random.mtrand.RandomState()
+
+
+def tensor_table_from_df(df: DataFrameType):
+    from merlin.table import CupyColumn, NumpyColumn, TensorTable
+
+    array_cols = {}
+    column_type = NumpyColumn if isinstance(df, pd.DataFrame) else CupyColumn
+    device = "cpu" if isinstance(df, pd.DataFrame) else None
+
+    for col in df.columns:
+        if is_list_dtype(df[col]):
+            values_series, offsets_series = pull_apart_list(df[col], device=device)
+            array_cols[col] = column_type(values_series.values, offsets_series.values)
+        else:
+            array_cols[col] = column_type(df[col].values)
+
+    return TensorTable(array_cols)
+
+
+def df_from_tensor_table(table):
```

Comment:
If we want to support TensorTable conversions to DataFrames, Would it be easier to discover this functionality for users by making a call to this from a method like `tensor_table.to_dataframe()`?

Reply:
Yeah, it would be easier to discover that way. One of the wrinkles that led us down this path is that we didn't want to make `TensorTable` itself depend on Pandas/cuDF`. `dispatch.py` is where we currently handle operations that depend on dataframe types and whether or not `cudf` is available, so we went with that for now. I suppose we could put the dataframe library imports directly in these methods to avoid importing them everywhere `TensorTable` is used.

Reply:
adding the imports into the methods would be a reasonable option if we wanted to add these methods. We could do that in a follow-up PR. If we added a `to_dataframe` method, it would also be nice to have a corresponding `from_dataframe` classmethod

Reply:
That's the style I like too. 👍🏻 

An open question for me is how we want the methods for converting across frameworks at the table level to look. With dtype conversion we did `dtype.to("numpy")`, which works well when some types or frameworks may not be available but has the downside of being "stringly typed." Unless we come up with something better, I'd probably use that here too, but don't feel entirely great about it.

Reply:
If the framework column classes are intended to be part of the public API, then could use those instead of strings? something along the lines of `table.convert_columns_to(TorchColumn)` or maybe something more concise. Since we don't need this functionality in the immediate term in systems we can figure this out in a follow-up change